scripts/migrations/olids_2026/rewrite_transitive.py

```
@@ -0,0 +1,263 @@
+    name_to_path: dict[str, pathlib.Path] = {}
+    path_to_refs: dict[pathlib.Path, set[str]] = {}
+    for root in scan_paths:
+        if not root.exists():
+            continue
+        for p in root.rglob("*.sql"):
+            stem = p.stem
+            name_to_path.setdefault(stem, p)
+            try:
+                text = p.read_text(encoding="utf-8", errors="replace")
+            except Exception:
+                continue
```

Ruff (0.15.13)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)